azurerm/internal/services/policy/policy_assignment_resource.go

```diff
@@ -123,10 +126,40 @@ func resourceArmPolicyAssignment() *schema.Resource {
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+
+			"metadata": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: policyAssignmentsMetadataDiffSuppressFunc,
+			},
 		},
 	}
 }
 
+func policyAssignmentsMetadataDiffSuppressFunc(_, old, new string, _ *schema.ResourceData) bool {
+	var oldPolicyAssignmentsMetadata map[string]interface{}
+	errOld := json.Unmarshal([]byte(old), &oldPolicyAssignmentsMetadata)
+	if errOld != nil {
+		return false
+	}
+
+	var newPolicyAssignmentsMetadata map[string]interface{}
+	errNew := json.Unmarshal([]byte(new), &newPolicyAssignmentsMetadata)
+	if errNew != nil {
+		return false
+	}
```

Comment:
This will need to be wrapped as `new` will be an empty string if `metadata` is absent from the config causing `Unmarshal` to return an error since that wouldn't be valid JSON.
```suggestion
	if new != "" {
		errNew := json.Unmarshal([]byte(new), &newPolicyAssignmentsMetadata)
		if errNew != nil {
			return false
		}
	}
```